worker/cmd/generate.go

```diff
@@ -196,15 +212,67 @@ var generateCmd = &cobra.Command{
 		go func(ch <-chan string) {
 			defer wg.Done()
 			for job := range ch {
-				jp := NewJobProcessor(ctx, pool, svc, sugar, job, PreCheckEndpointURL, SdgEndpointURL, TlsClientCertPath, TlsClientKeyPath, TlsServerCaCertPath, MaxSeed)
+				jp := NewJobProcessor(ctx, pool, svc, sugar, job, PreCheckEndpointURL, SdgEndpointURL, TlsClientCertPath, TlsClientKeyPath, TlsServerCaCertPath, MaxSeed, CleanupStrategy)
 				jp.processJob()
 			}
 		}(jobChan)
 
+		wg.Add(1)
+		go func(ch <-chan DiskPressureEvent, stopChan <-chan struct{}) {
+			defer wg.Done()
+			timer := time.NewTicker(5 * time.Minute)
+			for {
+				select {
+				case <-stopChan:
+					sugar.Info("Shutting down disk-pressure listener")
+					close(jobChan)
+					return
+				case <-timer.C:
+					diskPressure, err := DiskUsage()
+					var pressureEvent DiskPressureEvent
+					if err != nil {
+						sugar.Errorf("Could not get disk usage: %v", err)
+						continue
+					}
+					if diskPressure < 33 {
+						pressureEvent = lowDiskPressureEvent
+					} else if diskPressure < 66 {
+						pressureEvent = mediumDiskPressureEvent
+					} else {
+						pressureEvent = highDiskPressureEvent
+
```

Comment:
Maybe I could set the strategy to `immediate` if it hits `highDiskPressureEvent`...?